packages/lambda-tiler/src/wmts.capability.ts

```diff
@@ -288,26 +339,64 @@ export class WmtsCapabilities {
       }),
     ]);
   }
+
   toVNode(): VNodeElement {
-    const layers: (VNodeElement | null)[] = [];
-    layers.push(this.buildLayer(this.tileSet));
-
-    if (this.layers) {
-      const layerByName = new Map<string, ConfigLayer>();
-      // Dedupe the layers by unique name
-      for (const img of this.layers) layerByName.set(standardizeLayerName(img.name), img);
-      const orderedLayers = Array.from(layerByName.values()).sort((a, b) =>
-        (a.title ?? a.name).localeCompare(b.title ?? b.name),
-      );
-      for (const img of orderedLayers) layers.push(this.buildLayerFromImagery(img));
-    }
+    // Prepare provider vNode if exists
+    const provider = this.toProviderVNode(this.provider);
+
+    // Build TileSet Layer VNodes
+    const layers: VNodeElement[] = [];
+    layers.push(this.toLayerVNode(this.tileSet));
+    const contents = layers.concat(this.toAllImageryLayersVNode(this.configLayers));
+
+    // Build TileMatrix Sets vNodes
+    for (const tms of this.tileMatrixSets.values()) contents.push(this.buildTileMatrixSet(tms));
+
+    return V('Capabilities', CapabilitiesAttrs, [...provider, V('Contents', contents)]);
+  }
+}
 
-    for (const tms of this.tileMatrixSets.values()) layers.push(this.buildTileMatrixSet(tms));
+export interface WmtsCapabilitiesParams {
+  provider?: WmtsProvider;
+  /** Tileset to export into WMTS */
+  tileSet: ConfigTileSet;
+  /** List of tile matrixes to output */
+  tileMatrix: TileMatrixSet[];
+  /** All the imagery used by the tileSet and tileMatrixes */
+  imagery: Map<string, ConfigImagery>;
+  /** Limit the output to the following image formats other wise @see ImageFormatOrder */
+  formats?: ImageFormat[];
+  /** Specific layers to add to the WMTS */
+  layers?: ConfigLayer[];
+}
 
-    return V('Capabilities', CapabilitiesAttrs, [...this.buildProvider(), V('Contents', layers)]);
+/**
+ * WMTS Capabilities Builder
+ *
+ * /v1/tiles/:tileSet/:tileMatrix/WMTSCapabilities.xml
+ * /v1/tiles/:tileSet/WMTSCapabilities.xml
+ * /v1/tiles/WMTSCapabilities.xml
+ *
+ * @example
+ *
+ */
+export class WmtsCapabilities {
+  builder: WmtsCapabilitiesBuilder;
+
+  constructor(builder: WmtsCapabilitiesBuilder, params: WmtsCapabilitiesParams) {
+    this.builder = builder;
+    // Set Builder variables
+    this.builder.addTileMatrix(params.tileMatrix);
+    this.builder.addImagery(params.imagery);
+    this.builder.addFormats(params.formats);
+
+    // Build wmts capabilities
+    this.builder.addTileSet(params.tileSet);
```

Comment:
`addX` needs to be able to be called multiple times, eg

```typescript
x.addFormats(ImageFormat.WebP)
x.addFormats(ImageFormat.Avif)
```

another option there would be to support

```typescript
x.addFormat(ImageFormat.WebP, ImageFormat.Avif)
x.addFormat(...params.formats)
```

should add both webp and avif

```typescript
for (const im of imagery) x.addImagery(im)
```